sdk/storage/inc/blobs/internal/protocol/blob_rest_client.hpp

```diff
@@ -0,0 +1,2630 @@
+                  blob_container_item = BlobContainerItem();
+                else if (type == end_tag && name == "Container")
+                  response.BlobContainerItems.emplace_back(std::move(blob_container_item));
+                else if (type == content && name == "Name")
+                  blob_container_item.Name = value;
+                else if (type == content && name == "Last-Modified")
+                  blob_container_item.LastModified = value;
+                else if (type == content && name == "Etag")
+                  blob_container_item.ETag = value;
+                else if (type == content && name == "LeaseStatus")
+                  blob_container_item.LeaseStatus = BlobLeaseStatusFromString(value);
+                else if (type == content && name == "LeaseState")
+                  blob_container_item.LeaseState = BlobLeaseStateFromString(value);
+                else if (type == content && name == "LeaseDuration")
+                  blob_container_item.LeaseDuration = value;
+                else if (type == content && name == "PublicAccess")
+                  blob_container_item.AccessType = PublicAccessTypeFromString(value);
+                else if (type == content && name == "HasImmutabilityPolicy")
+                  blob_container_item.HasImmutabilityPolicy = value == "true";
+                else if (type == content && name == "HasLegalHold")
+                  blob_container_item.HasLegalHold = value == "true";
+              };
+
+        std::function<void(xmlNode*)> parse_xml;
+        parse_xml = [&parse_xml, &parse_xml_callback](xmlNode* node) {
+          if (!(node->type == XML_ELEMENT_NODE || node->type == XML_ATTRIBUTE_NODE))
+            return;
+
+          std::string node_name(reinterpret_cast<const char*>(node->name));
+          parse_xml_callback(node_name, start_tag, "");
+
+          for (xmlAttr* prop = node->properties; prop; prop = prop->next)
+          {
+            std::string prop_name(reinterpret_cast<const char*>(prop->name));
+            std::string prop_value(reinterpret_cast<const char*>(prop->children->content));
+            parse_xml_callback(prop_name, attribute, prop_value);
+          }
+
+          bool has_child_element = false;
+          for (xmlNode* child = node->children; child; child = child->next)
+          {
+            has_child_element |= child->type == XML_ELEMENT_NODE;
+            parse_xml(child);
+          }
+
+          if (!has_child_element && node->children)
+          {
+            std::string node_content(reinterpret_cast<const char*>(node->children->content));
+            parse_xml_callback(node_name, content, node_content);
+          }
+
+          parse_xml_callback(node_name, end_tag, "");
+        };
+
+        parse_xml(root);
+
+        xmlFreeDoc(doc);
+
+        // TODO: Think about how to cleanup
+        // xmlCleanupParser();
+        return response;
+      }
+
+      static ListContainersSegment ListBlobContainers(
```

Comment:
sorry, I didn't get you. Why it's a subset? are you talking about `Content-Type` http respone header?

Reply:
If you're talking about `Content-Type` header, i'm okay if you want to add it. Actually on a second thought, I think it also makes some sense to return it to end user. Because the HTTP body is returned to end user (not for now because we don't have the Response template class). The end user may want to know if the body is xml or json or something else.